src/kudu/consensus/log_cache.cc

```diff
@@ -164,17 +197,182 @@ void LogCache::TruncateOpsAfterUnlocked(int64_t index) {
   next_sequential_op_index_ = index + 1;
 }
 
+Status LogCache::UncompressMsg(const ReplicateRefPtr& msg,
+                               faststring& buffer,
+                               std::unique_ptr<ReplicateMsg>* uncompressed_msg) {
+  const auto op_type = msg->get()->op_type();
+  const auto& payload = msg->get()->write_payload();
+  const auto compression_codec = payload.compression_codec();
+  const auto uncompressed_size = payload.uncompressed_size();
+  const auto compressed_size = payload.payload().size();
```

Comment:
nit: it helps the code reader to see the actual types instead of auto if the types are not too complicated.